yt/data_objects/static_output.py

```diff
@@ -1879,7 +1880,13 @@ def _reconstruct_ds(*args, **kwargs):
 
 
 @functools.total_ordering
-class ParticleFile:
+class ParticleFile(abc.ABC):
+    filename: str
+    file_id: int
+    start: int = None
+    end: int = None
+    total_particles: ParticleTypeSizes = None
```

Comment:
These won't type check, you need to `from typing import Optional`
```suggestion
    start: Optional[int] = None
    end: Optional[int] = None
    total_particles: Optional[ParticleTypeSizes] = None
```
furthermore, it looks to me that the underlying logic can be simplified a lot in `__init__`. `start` doesn't need a `None` default value, the real default is `0`, right now it's equivalent but with extra steps.

Reply:
Hmm, they aren't optional though.  Is your suggestion still applicable?

Reply:
Or should I leave them without initialization?

Reply:
`typing.Optional` strictly means that `None` is an acceptable value despite the explicit type being incompatible.
Omitting it is discouraged by PEP 484
>A past version of this PEP allowed type checkers to assume an optional type when the default value is None, as in this code:
>```python
>def handle_employee(e: Employee = None): ...
>```
>This would have been treated as equivalent to:
>```python
>def handle_employee(e: Optional[Employee] = None) -> None: ...
>```
>This is no longer the recommended behavior. Type checkers should move towards requiring the optional type to be made explicit.

source:
https://www.python.org/dev/peps/pep-0484/#union-types

Reply:
So please keep the default None value, but use `Optional` too, otherwise we'll need to change add it later when we perform static type checking.

Reply:
you might want to try: https://pypi.org/project/auto-optional/

Reply:
OK, I will think on this and work through the process of coming around to realizing you are correct... :)